content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md

```diff
@@ -0,0 +1,261 @@
+
+Run the following commands to create the role and verify that the role has been created:
+
+```bash
+$ kubectl apply -f role.yaml 
+role.rbac.authorization.k8s.io/cfe-demo-role created
+
+$ kubectl get role cfe-demo-role 
+NAME            CREATED AT
+cfe-demo-role   2022-05-19T20:51:57Z
+```
+
+### Grant Permissions to Service Account
+You now create the RoleBinding to bind the role to the service account. 
+
+Here is the manifest file to create the RoleBinding that binds the role `cfe-demo-role` to the service account `cfe-demo-sa`. Replace those names with the names in your environment.
+
+
+```json
+# rolebinding.yml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: cfe-demo-rb
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: cfe-demo-role # Should match name of Role
+subjects:
+- kind: ServiceAccount
+  name: cfe-demo-sa # Should match service account name
+```
+Run the following commands to create the rolebinding and verify that the rolebinding has been created:
+
+
+```bash
+$ kubectl apply -f rolebinding.yaml 
+rolebinding.rbac.authorization.k8s.io/cfe-demo-rb created
+
+$ kubectl get rolebindings cfe-demo-rb 
+NAME          ROLE                 AGE
+cfe-demo-rb   Role/cfe-demo-role   19s
+```
+
+### Extract Service Account Token
+You check the secret token in the created service account and extract the token field by running the following commands. The token is a randomized string. The setup shows only a snippet of this string.
+
+
+
+```bash
+$ kubectl describe serviceaccounts cfe-demo-sa 
+Name:                cfe-demo-sa
+Namespace:           cfe-demo-cluster
+Labels:              <none>
+Annotations:         Image pull secrets:  <none>
+Mountable secrets:   cfe-demo-sa-token-2zlzf
+Tokens:              cfe-demo-sa-token-2zlzf
+Events:              <none>
+
+$ kubectl describe secrets cfe-demo-sa-token-2zlzf 
+Name:         cfe-demo-sa-token-2zlzf
+Namespace:    cfe-demo-cluster
+Labels:       <none>
+Annotations:  kubernetes.io/service-account.name: cfe-demo-sa
+              kubernetes.io/service-account.uid: a467f9bd-655d-413f-ad77-5156d03d2322
+
+Type:  kubernetes.io/service-account-token
+
+Data
+====
+ca.crt:     1066 bytes
+namespace:  16 bytes
+token:      iIsImtpZCI6IjA2YnhmSVZrVDRGWnBab0VOYXhnWFBTTE1WWmptUm40eER
+
+```
+
+Note that if you use `-o yaml` instead of `describe` in the commands, you get a base64-encoded version of the token. You must decode it before you use it.
+
+
+
+If you access the Kubernetes API directly, e.g., from `curl`, you can use the token as the bearer token for the authorization header. 
+
+However, if you have your scripts running outside the cluster that use kubectl or a client library to access the Kubernetes cluster, you need the kubeconfig file to load configs from. You follow up the following section to create such a kubeconfig file.
```

Comment:
Change "cluster that use kubectl" to "cluster that uses kubectl"
Change "to load configs from" to "to load the configs from"
Change the last sentence to read "Follow the instructions in the next section to create such a kubeconfig file."